src/EventSubscriber/TraceSubscriber.php

```diff
@@ -69,8 +76,26 @@ public function onResponse(ResponseEvent $event): void
             return;
         }
 
-        if ($this->sendResponseHeader) {
+        $request   = $event->getRequest();
+        $sendHeader = $this->sendResponseHeader && $this->isTrustedRequest($request, $this->trustedIpsResponse);
+        if ($sendHeader) {
             $this->traceService->handleResponse($event->getResponse(), $this->traceStorage->getTrace());
         }
     }
+
+    /**
+     * @param string[]|string|null $trustedIps
+     */
+    private function isTrustedRequest(Request $request, array|string|null $trustedIps): bool
+    {
+        if ($trustedIps === null) {
+            return true;
+        }
+
+        if (is_string($trustedIps)) {
+            $trustedIps = array_map('trim', explode(',', $trustedIps));
```

Comment:
Hidden feature? The documentation doesn't describe that comma separated string is supported. Is there a phpstan `comma-separated-string` type?

Reply:
Nice addition for the future might also be mask support. `198.0.0.0/24`

Reply:
Yes, same input is supported as Symfony's trustedProxies configuration. Using the same way to validate the IP as Symfony does.
Already supports ranges + masks.